src/System.Management.Automation/engine/runtime/Operations/MiscOps.cs

```diff
@@ -161,7 +162,12 @@ internal static class PipelineOps
             commandProcessor.UseLocalScope = !dotSource &&
                                              (cmd is ScriptCommand || cmd is PSScriptCmdlet);
 
-            bool isNativeCommand = commandProcessor is NativeCommandProcessor;
+            bool isNativeCommand = false;
+            if (commandProcessor is NativeCommandProcessor)
+            {
+                isNativeCommand = true;
+            }
```

Comment:
This change can be reverted
```suggestion
            bool isNativeCommand = commandProcessor is NativeCommandProcessor;
```

Reply:
Done!